### PR TITLE
Add --replace-backend-flags to cargo-verify

### PR DESCRIPTION
**Pull request by fshaked**
Also add Handlebar templates to --backend-flags.

### CODE REVIEW

**Comment by fshaked** [comment]
@alastairreid, note that --replace-backend-flags doesn't pass `Opt::args` to the verified program, because there is no simple way to do it (will need a handlebar template that expands to Vec<String>, or something like that). The user should pass those through --backend-flags instead, if she uses --replace-backend-flags.

**Comment by alastairreid** [comment]
Just before I dive into the code...

Passing args to the program is a really useful capability - it's fundamental for finding bugs in standalone programs like uutils/coreutils.
So I would **really** not like to lose that capability.

File: cargo-verify/src/seahorn.rs [hunk 1]
```diff
@@ -78,39 +78,46 @@ fn importance(line: &str, expect: &Option<&str>, name: &str) -> i8 {
...
-        .args(&opt.backend_flags)
-        .arg(&bcfile);
-    // .args(&opt.args)
+
+    let user_flags: Vec<_> = opt.backend_flags.iter().map(|flag| {
+        backends_common::format_flag(&flag, &entry, &bcfile, &out_dir)
+    }).collect::<Result<_,_>>()?;
+
+    if ! opt.replace_backend_flags {
+        cmd.args(&["bpf",
```
Comment (alastairreid) [inline cargo-verify/src/seahorn.rs:87]: I thought that the plan was to be able to provide this content as a yaml file?

**Review by alastairreid — COMMENTED** [review]
My main thought when reviewing a change like this is "how does this change the user interface?"
So it would be good if this change also included tests or changes to existing tests that show the necessary changes.

**Comment by alastairreid** [comment]
Following on from my previous comment...
If this changes how arguments are passed to programs, does this require a change to scripts/regression-tests or do those tests (especially of demos/simple/argv) still pass?

File: cargo-verify/src/seahorn.rs (hunk 1 above)
Comment (fshaked) [reply cargo-verify/src/seahorn.rs:87]: As you already noted in #119, this was implemented separately.

**Comment by fshaked** [comment]
> If this changes how arguments are passed to programs, does this require a change to scripts/regression-tests or do those tests (especially of demos/simple/argv) still pass?

This PR doesn't change anything if you don't use `--replace-backend-flags` on the command line. So script/regression-tests should be unaffected.

I will add to regression-tests something that uses `--replace-backend-flags`.

**Pull request merged**